internal/event/dispatch_service.go

```diff
@@ -68,9 +68,13 @@ func NewDispatchService(logger ilogger.LogAdaptor) *DispatchService {
 
 // Stop the DispatchService and make sure that all the successful Publish events are handled before close
 func (s *DispatchService) Stop(ctx context.Context) error {
-	if !atomic.CompareAndSwapInt32(&s.state, ready, stopped) {
+	s.subscriptionsMu.Lock()
+	if s.state == stopped {
+		s.subscriptionsMu.Unlock()
 		return nil
 	}
+	s.state = stopped
+	s.subscriptionsMu.Unlock()
```

Comment:
Could you revert this change? `subscriptionMu` is used for subscribe/unsubscribe/publish.

Reply:
It is correct but it is required a protection mechanism for state changes, otherwise test result becomes inconsistent.